identifier/vite.config.js

```diff
@@ -1,6 +1,41 @@
-import { defineConfig, splitVendorChunkPlugin } from 'vite';
-import react from '@vitejs/plugin-react';
-import checker from 'vite-plugin-checker';
+import { defineConfig, splitVendorChunkPlugin } from "vite";
+import react from "@vitejs/plugin-react";
+import checker from "vite-plugin-checker";
+import legacy from "@vitejs/plugin-legacy";
+import { readFileSync, writeFileSync } from "fs";
+import path from "path";
+
+function cspPlugin() {
+  return {
+    name: "csp-plugin",
+    closeBundle() {
+      // Path to the generated HTML file after build
+      const htmlFilePath = path.resolve(__dirname, "build/index.html");
+
+      // Read the HTML file content
+      let htmlContent = readFileSync(htmlFilePath, "utf-8");
```

Comment:
Please study the vite plugin API. Especially https://vite.dev/guide/api-plugin#transformindexhtml - there is no need to implement things like loading the index html manually.

Reply:
Hi @longsleep , I did went through vite plugin api and had the initial idea of using it like this. 

![image](https://github.com/user-attachments/assets/58c1af2f-cfaf-42dd-9be4-176a390b2d83)

Problem is, this does not replace the data as I want, because even if the order is post, it still does not have the inline yet applied to the code.

![image](https://github.com/user-attachments/assets/64e4a9f3-cdce-4a3e-87d7-ffc10f6b758b)

As you can see, since the inline scripts are not yet applied to the html, it does not replaces it value.

This readFileSync, syncs with the up-to-date file and replaces the correct content I need.

Reply:
Please figure out why that is and use the plugin API as it is intended.

Reply:
Updated the vite config with transform html changes. No longer needed to use path or readFileSync.